### PR TITLE
CT-265 add statesman

### PR DESCRIPTION
**Pull request by misaka**
https://dsdmoj.atlassian.net/browse/CT-265

Replace existing state implementation with statesman. This is a bit more than just an exploration, as we need this implemented now, but we did investigate `state_machines`, `aasm`, `ssm`, etc as alternative gems. This one seems to require the least amount of modifications to make work for our needs.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -72,4 +74,3 @@ gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 
 # For Heroku
 gem 'rails_12factor', group: :production
```
Comment (aliuk2012) [inline Gemfile]: should just remove this line as well because we will not be needing it

**Comment by aliuk2012** [comment]
Has this taken into account all these statuses and copy for each status? https://dsdmoj.atlassian.net/wiki/display/CD/Spec+manual+-+Case+progression

Apart from the copy on the case list and case detail page its looks good. I think we should maybe try include the correct status copy as part of this PR.

**Comment by EdwardAndress** [comment]
@aliuk2012 and @misaka - I have added one more commit to address Al's comment about copy.  The copy was approved by @kelliematheson

**Comment by EdwardAndress** [comment]
We don't explicitly test that all transitions are saved, when several changes are made to a case.  For example, for a case that has been created, assigned and accepted, we might expect to see 2 or 3 transitions, with the correct details, in a particular order.

I suspect that we can cover this when we implement the case history though.

**Comment by EdwardAndress** [comment]
@misaka - we should probably all review this one.

File: app/views/cases/edit.html.slim [hunk 2]
```diff
@@ -14,4 +14,4 @@ section.question.status
   h2
     = 'Status'
   p
-    = @case.state.humanize
+    = @case.current_state.humanize
```
Comment (EdwardAndress) [inline app/views/cases/edit.html.slim]: :) Yes, well spotted.

**Comment by misaka** [comment]
@EdwardAndress re: testing that all transitions are saved, I'm not sure what you mean. Are you saying that if we create a case in a specific state (e.g. `drafting`), we should also create the transitions? That doesn't make sense because A) we can't set the `state` when we create the case, we've removed that column, it's now inferred from `case_transitions`. B) we wouldn't have the necessary information (`assignee_id`, and in the case of rejection `message`, etc).

**Comment by EdwardAndress** [comment]
@aliuk2012 and @misaka - this is pretty much ready but please have one more look through.

**Comment by misaka** [comment]
@EdwardAndress Looking good. Just waiting for @aliuk2012 to check a couple things and I'll squash and merge.

**Pull request merged**